src/components/ec-text-field/ec-text-field.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="ec-text-field">
+    <div
+      v-if="label"
+      class="ec-text-field__label"
+    >{{ label }}</div>
+    <input
+      class="ec-text-field__input"
+      :class="{
+        'ec-text-field__input--error': state === 'error',
+      }"
+      :type="type"
+      :value="value"
+      @input="(e) => $emit('change', e.target.value)"
+    >
+    <div
+      v-if="state === 'error'"
+      class="ec-text-field__error-label"
+    >{{ errorMessage }}</div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'EcTextField',
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    type: {
+      type: String,
+      default: 'text',
+      validator(value) {
+        return ['text', 'date', 'number'].includes(value);
+      },
+    },
+    value: {
+      default: '',
+      type: [Number, String, Date],
+    },
+    label: {
+      default: '',
+      type: String,
+    },
+    errorMessage: {
+      default: '',
+      type: String,
+    },
+    state: {
```

Comment:
I strongly disagree with the name of this prop. A `state` of the component is its internal state, a `vm`. It's very misleading to have a property with name like that, it indicates you can pass the entire state of the component from the parent, which is a big anti-pattern.

The prop can only have two values, so it would be better to make it Boolean and call it `has-error`.
Even better it would be to get rid of it and rather check if `errorMessage` has been passed. If errorMessage is empty, input is valid, otherwise make it invalid.

Reply:
I did in that way thinking about the future because we will need status success, and about the errorMessage, I thought it will better to set the message and play with the status better than set and unset the message of the error

Reply:
let's name this status then, or downsize to just error as suggested by Milos